### ISSUE TITLE
[Question] Separate endpoints for cookie and token based clients?

### ISSUE THREAD

**Issue by j-jalving** [manchenkoff/nuxt-auth-sanctum#120]
I'm trying to implement nuxt-auth-sanctum for my Nuxt/Capacitor project that will be deployed to the web and as a native app. For my web build I will use cookie based authentication, but for my native app build I will use token based authentication.

The documentation tells me:
```
Your API should have at least two endpoints for login and logout which are not covered by stateful middleware, so make sure that you do not use the same endpoints as for cookie-based authentication to avoid CSRF token mismatch errors.
```

Does this only go for the login and logout endpoints? Or should all my endpoints ('forgot-password', 'user', etc.) have separate versions for clients that use token based authentication?

**Comment by manchenkoff** [comment]
Hey @j-jalving, have you managed to test your mobile app authentication?

**Comment by j-jalving** [comment]
Hi, yeah I've been meaning to get back to you. The issues I encountered were due to Laravel Sanctum including the authentication routes in the web.php route file by default. Moving these routes to the api.php route file ensures that the correct middleware is applied.

Sanctum only applies stateful middleware for requests originating from domains listed in SANCTUM_STATEFUL_DOMAINS. Therefore, when using cookie-based authentication, set this to the domain of your *web* app. Since the referrer for a Capacitor *native* app is `capacitor://localhost`, stateful middleware will not be applied for those requests. CSRF protection seems to be working fine in both cases without having manually excluded any routes from CSRF middleware.

The main reason for having separate mobile login and logout endpoints is to modify them for token-based authentication and allowing the `/mobile/login` (and possibly `/mobile/register`) to return a token in the response.

I'm really digging you package, clean code and documentation and so far it seems to be working perfectly!
